? faceted_search_php53fixes.patch
Index: faceted_search.inc
===================================================================
RCS file: /cvs/drupal-contrib/contributions/modules/faceted_search/faceted_search.inc,v
retrieving revision 1.57
diff -u -p -r1.57 faceted_search.inc
--- faceted_search.inc	4 Jan 2009 19:36:25 -0000	1.57
+++ faceted_search.inc	20 Feb 2010 00:26:01 -0000
@@ -901,7 +901,11 @@ class faceted_search {
     $this->settings['types'] = array();
 
     // Provide other modules an opportunity to add their own default settings.
-    module_invoke_all('faceted_search_init', $this);
+    $hook = 'faceted_search_init';
+    foreach (module_implements($hook) as $module) {
+      $function = $module .'_'. $hook;
+      $function($this);
+    }
   }
 
   /**
@@ -1162,7 +1166,11 @@ class faceted_search {
 
     // Give other modules an opportunity at altering the final query (e.g. for
     // additional filtering).
-    module_invoke_all('faceted_search_query_alter', $this, $query);
+    $hook = 'faceted_search_query_alter';
+    foreach (module_implements($hook) as $module) {
+      $function = $module .'_'. $hook;
+      $function($this, $query);
+    }
 
     // Perform the search results query and store results in a temporary table.
     //
@@ -1241,7 +1249,11 @@ class faceted_search {
       // There is no results table at this point, so we can't rely on the
       // results table having been filtered already. Therefore, we ask modules
       // to alter the categories query instead.
-      module_invoke_all('faceted_search_query_alter', $this, $query);
+      $hook = 'faceted_search_query_alter';
+      foreach (module_implements($hook) as $module) {
+        $function = $module .'_'. $hook;
+        $function($this, $query);
+      }
     }
     elseif ($this->_results_count > 0) {
       // Search within results.
